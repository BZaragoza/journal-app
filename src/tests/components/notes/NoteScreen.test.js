import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import NoteScreen from '../../../components/notes/NoteScreen';
import { activeNote } from '../../../actions/notes';



jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn()
}));

const mockStore = configureStore([thunk]);

const initState = {
  auth: {
    uid: '1',
    name: 'Brian'
  },
  ui: {
    loading: false,
    msgError: null
  },
  notes: {
    notes: [],
    active: {
      id: 1234,
      title: "",
      body: ""
    }
  }
}

const store = mockStore( initState );
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={ store }>
    <NoteScreen />
  </Provider>
)

describe('Tests on <NoteScreen />', () => {
  
  test('should render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should trigger activeNote', () => {
    
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: "title",
        value: 'Changed'
      }
    });

    expect( activeNote ).toHaveBeenCalledWith( 1234, {"body": "", "id": 1234, "title": "Changed"});

  })
  

  

})
