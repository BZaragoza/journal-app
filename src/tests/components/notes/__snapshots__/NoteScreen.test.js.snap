// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <NoteScreen /> should render correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      08 de Junio de 2021
    </span>
    <input
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value=""
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today?"
      value=""
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
